### PR TITLE
PartiallyEvaluateExpression: Avoid unrecognized paramkind: 2

### LINKED ISSUE #3624: unrecognized paramkind: 2 when UPDATE has subquery using volatile function in WHERE if table created using master_create_distributed_table
```
CREATE TABLE articles_hash (
        id bigint NOT NULL,
        author_id bigint NOT NULL,
        title varchar(20) NOT NULL,
        word_count integer
);
SELECT master_create_distributed_table('articles_hash', 'author_id', 'hash');

UPDATE articles_hash SET id = id
WHERE title IN (SELECT name FROM authors_reference WHERE random() > 0.5);
```

Spits out "ERROR:  unrecognized paramkind: 2"

Changing `master_create_distributed_table('articles_hash', 'author_id', 'hash')` to `create_distributed_table('articles_hash', 'author_id')` fixes the issue

Including `author_id = 1` in WHERE has same result

### PR DESCRIPTION
**Pull request by serprex**
Fixes #3624

### CODE REVIEW

File: src/backend/distributed/utils/citus_clauses.c [hunk 1]
```diff
@@ -27,7 +27,7 @@
 
 
 /* private function declarations */
-static bool IsVarNode(Node *node);
+static bool CannotCitusEvaluate(Node *node);
```
Comment (SaitTalhaNisanci) [inline src/backend/distributed/utils/citus_clauses.c]: What does CannotCitusEvaluate mean?

Comment (serprex) [reply src/backend/distributed/utils/citus_clauses.c]: CannotCitusEvaluate returns whether citus_evaluate_expr cannot evaluate node

Comment (SaitTalhaNisanci) [reply src/backend/distributed/utils/citus_clauses.c]: Maybe using CanCitusEvaluateExpression is easier to understand

Comment (marcocitus) [reply src/backend/distributed/utils/citus_clauses.c]: Maybe just `IsVarOrParamSublink`. If we need to add another condition, we can come up with a new name.

File: src/backend/distributed/utils/citus_clauses.c [hunk 2]
```diff
@@ -120,7 +120,7 @@ PartiallyEvaluateExpression(Node *expression,
 	else if (ShouldEvaluateExpressionType(nodeTag) &&
 			 ShouldEvaluateFunctionWithMasterContext(masterEvaluationContext))
 	{
-		if (FindNodeCheck(expression, IsVarNode))
+		if (FindNodeCheck(expression, CannotCitusEvaluate))
```
Comment (marcocitus) [inline src/backend/distributed/utils/citus_clauses.c]: Maybe add a comment on why this check is needed.

**Pull request merged**